### PR TITLE
runtime: Update neomutt to latest release

### PR DESCRIPTION
**Pull request by flatcap**
- Update `syntax/neomuttrc.vim` to the latest version (NeoMutt 2024-10-02)
- Add `syntax/neomuttlog.vim` to handle NeoMutt log files

The first 3 commits perform some tidying up of `neomuttrc.vim`
They make **no functional changes**.

- fe5bb737c neomutt: reorder some definitions
- c9c180609 neomutt: tidy whitespace
- 1fd9fe3b8 neomutt: rename some symbols
- 015651956 neomutt: update to release 2024-10-02
- ce9a3f93d neomutt: add neomuttlog syntax

Thanks to @avidseeker

### CODE REVIEW

File: runtime/doc/syntax.txt [hunk 1]
```diff
@@ -2524,6 +2524,12 @@ set "msql_minlines" to the value you desire.  Example: >
 	:let msql_minlines = 200
 
 
+NEOMUTT			*neomutt.vim* *ft-neomuttrc-syntax* *ft-neomuttlog-syntax*
+
+To use Neomutt default log colors instead of your colorscheme's >
+
+	:let g:neolog_default_colors = 1
```
Comment (chrisbra) [inline runtime/doc/syntax.txt]: Doesn't that rather mean, if it is not set, there won't be any coloring at all? Or what color schemes define those neolog groups? If that is the case, wouldn't it make sense to drop this variable and have it always defined?

Comment (flatcap) [reply runtime/doc/syntax.txt]: Hmm...
The colours look OK with a dark background, but horrible on a light background.

Would you prefer:
1. Always enabled
1. Invert the test to `g:neolog_disable_default_colors`

Comment (chrisbra) [reply runtime/doc/syntax.txt]: then I would invert the test please

Comment (flatcap) [reply runtime/doc/syntax.txt]: Thanks!  Done -- 7f4497cb8

**Comment by chrisbra** [comment]
thanks!

File: src/testdir/test_filetype.vim [hunk 2]
```diff
@@ -525,7 +525,7 @@ def s:GetFilenameChecks(): dict<list<string>>
     nanorc: ['/etc/nanorc', 'file.nanorc', 'any/etc/nanorc'],
     natural: ['file.NSA', 'file.NSC', 'file.NSG', 'file.NSL', 'file.NSM', 'file.NSN', 'file.NSP', 'file.NSS'],
     ncf: ['file.ncf'],
-    neomuttrc: ['Neomuttrc', '.neomuttrc', '.neomuttrc-file', '/.neomutt/neomuttrc', '/.neomutt/neomuttrc-file', 'Neomuttrc', 'Neomuttrc-file', 'any/.neomutt/neomuttrc', 'any/.neomutt/neomuttrc-file', 'neomuttrc', 'neomuttrc-file'],
+    neomuttrc: ['Neomuttrc', '.neomuttrc', '.neomuttrc-file', '/.neomutt/neomuttrc', '/.neomutt/neomuttrc-file', 'Neomuttrc', 'Neomuttrc-file', 'any/.neomutt/neomuttrc', 'any/.neomutt/neomuttrc-file', 'neomuttrc', 'neomuttrc-file', 'neomuttlog'],
```
Comment (clason) [inline src/testdir/test_filetype.vim:528]: That doesn't look correct given the change to `filetype.vim`?

(CI agrees.)

Comment (chrisbra) [reply src/testdir/test_filetype.vim:528]: yeah right :/

Comment (flatcap) [reply src/testdir/test_filetype.vim:528]: By default the log files are `~/.neomuttdebug0`, `1`, `2`, `3`, `4`

Comment (chrisbra) [reply src/testdir/test_filetype.vim:528]: thanks, let me adjust it

**Pull request closed without merging**